File: c/sorting_algorithms/comparative_sorting_algorithms/bubbleSort.c
#include <stdio.h>
void bubbleSort(int arr[], int n) {
    int temp;
    for (int i = 0; i < n-1; i++) {
        int swapped = 0;
        for (int j = 0; j < n-i-1; j++) {
            if (arr[j] > arr[j+1]) {
                // Intercambiar
                temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
                swapped = 1;
            }
        }
        if (!swapped) break;
    }
}

void printArray(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

int main() {
    int arr[] = {64, 34, 25, 12, 22, 11, 90};
    int n = sizeof(arr)/sizeof(arr[0]);
    printf("\nArray original: \n");
    printArray(arr, n);
    bubbleSort(arr, n);
    printf("\nArray ordenado: \n");
    printArray(arr, n);
    printf("\n");
    return 0;
}

File: python/sorting_algorithms/comparative_sorting_algorithms/bubbleSort.py
def bubble_sort(arr):
    n = len(arr)
    for i in range(n-1):
        swapped = False
        for j in range(n-i-1):
            if arr[j] > arr[j+1]:
                # Intercambiar
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = True
        if not swapped:
            break

# Ejemplo de uso
arr = [64, 34, 25, 12, 22, 11, 90]
print(f"Array original: {arr}")
bubble_sort(arr)
print(f"Array ordenado: {arr}")